prow/plugins/bugzilla/bugzilla.go

```diff
@@ -597,22 +603,28 @@ func handle(e event, gc githubClient, bc bugzilla.Client, options plugins.Bugzil
 				} else {
 					response += " There are no allowed bug groups configured for this repo."
 				}
-				return comment(response)
+				return status, comment(response)
 			}
-			return nil
+			return status, nil
 		}
 	}
 	// merges follow a different pattern from the normal validation
 	if e.merged {
-		return handleMerge(e, gc, bc, options, log, allRepos)
+		err := handleMerge(e, gc, bc, options, log, allRepos)
```

Comment:
It was a lot of work to decide where the action was taken, so before I revert all of this to leave the signature (which itself is NOT correct change as the status will always be false for now, so metrics will be corrupted), please confirm that is what you want.

Reply:
Also, if you want to revert all of the plugins, please let me at least keep LGTM and approve using this mechanism for now, as my real purpose is to observe GH API rate limits and can't do this without this change.

Reply:
I think having an unused signature is fine for now, it doesn't corrupt anything - we just don't report data where it is not yet exposed. I do truly think we can be much more formal and abstract about taking action. I can see a small handful of ways where the GitHub client, the k8s client, the Bugzilla client, the Git client, etc - they can report whether or not they did something transparently to the plugin author having to track it and we do not impose brittle and labor-intensive tracking for status, unless the plugin wants to expose something that was *not* a delegate client action.

Reply:
> I can see a small handful of ways where the GitHub client,

I am sorry, maybe it's because Golang it's not my native language, but I do not see a clear, harmless way to do this. Context is implicit and would have to be re-updated every handler execution since it's basically a new case. There could be a function wrapper using reflect which is obscure or interface re-implementation with the wrapper which is also not nice (probably per plugin which is even worse). Finally, other params could be added to the function but it also seems pointless. I maybe lack Golang experience but I miss a clean solution. Do you have a specific candidate in mind?

Reply:
@jmguzik when you inspect `plugins.NewAgent()` you will notice that we are actually creating a number of client objects with lifetimes local to individual invocations of handlers for individual GitHub webhook payloads. For instance, when we use `clientAgent.GitHubClient.WithFields(logger.Data).ForPlugin(plugin)` we decorate the individual GitHub client with logging data (which PR number, event GUID, etc) and with the plugin name, so that we may set the user agent header for calls to GitHub made while handling this particular webhook for this particular plugin, so that we may trace API requests per plugin.

Since we already have clients (for GitHub, Owners, and Bugzilla by the looks) that are individually created for a specific handler call, I think it should be as simple as adding a new member to those `struct`s, setting to to `true` if some API action is taken, and extending the interface they fulfill so that the member may be read after the handler is called in order to provide the correct label to the metric. I'm happy to link to an example commit if you'd like to see what I mean.

Of course such an approach would only handle the "simple" cases of client usage, if we wanted plugin authors to be able to mark "action taken" when no API interaction occurred, we would nevertheless need the function signature changes you made in this PR. However, being able to auto-detect what I think may be the most common cases would surely be nice.

Reply:
@jmguzik I'm sorry - I attempted to make this comment in my [first review](https://github.com/kubernetes/test-infra/pull/24938#discussion_r796889639) as well but that was _also_ nowhere near the first day that you had opened this PR, and nevertheless I was not effective at communicating. My apologies for this.